File: Orrery.js
input.onGesture(Gesture.Shake, function () {
    state = 0
    music.magicWand.playUntilDone()
    light.showAnimation(light.rainbowAnimation, 500)
    period = [24, 62, 100, 188]
    planets = ["Mercury", "Venus", "Earth", "Mars"]
    pix = [light.rgb(10, 10, 10), Colors.Yellow, Colors.Green, Colors.Red]
    loc = [Math.randomRange(0, 9), Math.randomRange(0, 9), Math.randomRange(0, 9), Math.randomRange(0, 9)]
    counter = [0, 0, 0, 0]
    light.setAll(0x000000)
    showPlanets()
    state = 1
    orrery()
})
function orrery () {
    if (1 == state) {
        showPlanets()
        for (let p = 0; p <= 3; p++) {
            counter[p] = 1 + counter[p]
            if (period[p] <= counter[p]) {
                counter[p] = 0
                light.setPixelColor(loc[p], 0x000000)
                x = 1 + loc[p]
                loc[p] = x % 10
            }
        }
    }
}
function showPlanets () {
    for (let index = 0; index <= 3; index++) {
        light.setPixelColor(loc[index], pix[index])
    }
}
input.buttonA.onEvent(ButtonEvent.Click, function () {
    if (0 == state) {
        state = 1
        orrery()
    } else {
        state = 0
    }
})
input.buttonB.onEvent(ButtonEvent.Click, function () {
    state = 0
    pix = [light.rgb(Math.randomRange(0, 100), Math.randomRange(0, 100), Math.randomRange(0, 100)), light.rgb(Math.randomRange(0, 100), Math.randomRange(0, 100), Math.randomRange(0, 100)), light.rgb(Math.randomRange(0, 100), Math.randomRange(0, 100), Math.randomRange(0, 100)), light.rgb(Math.randomRange(0, 100), Math.randomRange(0, 100), Math.randomRange(0, 100))]
    period = [Math.randomRange(10, 30), Math.randomRange(30, 80), Math.randomRange(50, 100), Math.randomRange(100, 200)]
    state = 1
    orrery()
})
let x = 0
let state = 0
let counter: number[] = []
let loc: number[] = []
let pix: number[] = []
let planets: string[] = []
let period: number[] = []
period = [24, 62, 100, 188]
planets = ["Mercury", "Venus", "Earth", "Mars"]
pix = [light.rgb(20, 20, 20), Colors.Yellow, Colors.Green, Colors.Red]
loc = [Math.randomRange(0, 9), Math.randomRange(0, 9), Math.randomRange(0, 9), Math.randomRange(0, 9)]
counter = [0, 0, 0, 0]
state = 0
showPlanets()
state = 1
orrery()
forever(function () {
    if (1 == state) {
        orrery()
    }
})

File: orrery.py
from adafruit_circuitplayground import cp
import time
import random
import board


def setperiod():
   return [24, 62, 100, 188]

def rndperiod():
   return [random.randrange(20)+10,
        random.randrange(50)+30,
        random.randrange(80)+30,
        random.randrange(150)+90]

def setcolor():
    return [(16,16,16), (64,60,7), (0, 62, 0), (36, 0, 0)]

def rndcolor():
    return [(random.randrange(100),random.randrange(20),random.randrange(20)),
            (random.randrange(20),random.randrange(20),random.randrange(100)),
            (random.randrange(20),random.randrange(100),random.randrange(20)),
            (random.randrange(50),random.randrange(40),random.randrange(30))]


period = setperiod()
planets = ["Mercury", "Venus", "Earth", "Mars"]
pix = setcolor()

loc = [0, 0 ,0,0]
counter = [0,0,0,0]
def clearplanets(lcx):
    for i in range(10):
        cp.pixels[i] = (0,0,0)
def showplanets(lcx):

    for p in range(4):
        cp.pixels[loc[p]] = pix[p]
        
def mvplanet(planet):
    cp.pixels[loc[p]]=(0,0,0)
    loc[planet] = (loc[planet]+1)%10
    cp.pixels[loc[p]]=pix[p]

locate = 2 # start at Earth


showplanets(loc)
while True:
    showplanets(loc)
    if cp.button_a:
        period = setperiod()
        pix = setcolor()
        
    if cp.button_b:
        period = rndperiod()
        pix = rndcolor()

    for p in range(4):
        counter[p] = counter[p]+1
        print (str(planets[p])+":"+str(counter[p])+":"+str(loc[p]))
        if counter[p]>period[p]:
            mvplanet(p)
            counter[p]=0
            
#    time.sleep(.01)
    
